src/sqlancer/oceanbase/oracle/OceanBaseNoRECOracle.java

```diff
@@ -0,0 +1,204 @@
+        }
+        rs.close();
+        return secondCount;
+    }
+
+    private int getFirstQueryCount(List<OceanBaseExpression> tableList, OceanBaseExpression randomWhereCondition,
+            List<OceanBaseExpression> groupBys) throws SQLException {
+        OceanBaseSelect select = new OceanBaseSelect();
+        select.setGroupByClause(groupBys);
+        // SELECT COUNT(t1.c3) FROM t1 WHERE (- (t1.c2));
+        // SELECT SUM(count) FROM (SELECT ((- (t1.c2)) IS TRUE) as count FROM t1);;
+        OceanBaseAggregate aggr = new OceanBaseAggregate(
+                new OceanBaseColumnName(new OceanBaseSchema.OceanBaseColumn("*", OceanBaseSchema.OceanBaseDataType.INT, false, 0, false)),
+                OceanBaseAggregate.OceanBaseAggregateFunction.COUNT);
+        select.setFetchColumns(Arrays.asList(aggr));
+        select.setFromList(tableList);
+        select.setWhereClause(randomWhereCondition);
+        select.setSelectType(OceanBaseSelect.SelectType.ALL);
+        int firstCount = 0;
+        optimizedQueryString = OceanBaseVisitor.asString(select);
+        SQLQueryAdapter q = new SQLQueryAdapter(optimizedQueryString, errors);
+        SQLancerResultSet rs;
+        if (options.logEachSelect()) {
+            logger.writeCurrent(optimizedQueryString);
+        }
+        try {
+            rs = q.executeAndGet(state);
+        } catch (Exception e) {
+            throw new AssertionError(firstQueryString, e);
+        }
+        if (rs == null) {
+            return -1;
+        }
+        if (rs.next()) {
+            firstCount += rs.getLong(1);
+        }
+        rs.close();
+        return firstCount;
+    }
+  
+    private OceanBaseExpression getTrueExpr(OceanBaseExpression randomWhereCondition){
+        int i = new Random().nextInt(8);
+        OceanBaseUnaryPostfixOperation isTrue = new OceanBaseUnaryPostfixOperation(randomWhereCondition,OceanBaseUnaryPostfixOperation.UnaryPostfixOperator.IS_TRUE,false);
+
+        OceanBaseUnaryPostfixOperation isNotTrue = new OceanBaseUnaryPostfixOperation(randomWhereCondition,OceanBaseUnaryPostfixOperation.UnaryPostfixOperator.IS_TRUE,true);
+
+        OceanBaseUnaryPostfixOperation isFalse= new OceanBaseUnaryPostfixOperation(randomWhereCondition,OceanBaseUnaryPostfixOperation.UnaryPostfixOperator.IS_FALSE,false);
+
+        OceanBaseUnaryPostfixOperation isNotFalse= new OceanBaseUnaryPostfixOperation(randomWhereCondition,OceanBaseUnaryPostfixOperation.UnaryPostfixOperator.IS_FALSE,true);
+
+        OceanBaseUnaryPostfixOperation isNULL= new OceanBaseUnaryPostfixOperation(randomWhereCondition,OceanBaseUnaryPostfixOperation.UnaryPostfixOperator.IS_NULL,false);
+
+        OceanBaseUnaryPostfixOperation isNotNULL= new OceanBaseUnaryPostfixOperation(randomWhereCondition,OceanBaseUnaryPostfixOperation.UnaryPostfixOperator.IS_NULL,true);
+
+        OceanBaseExpression expr = OceanBaseConstant.createNullConstant(); 
+        switch(i){
+           case 0:        
```

Comment:
The hard-coded constants look quite error-prone. Perhaps you can add an enum instead, of which a random value is selected, and based on which you perform the switch (or where each enum value implements a method that you call)?